src/recovery/tally.rs

```diff
@@ -577,46 +584,21 @@ impl FragmentReplayer {
 
                 let (vote_cast, identifier, _) =
                     deconstruct_account_transaction(&transaction_slice)?;
-                let address = Identifier::from(identifier.clone())
-                    .to_address(chain_addr::Discrimination::Production);
-
-                let choice = if let Payload::Public { choice } = vote_cast.payload() {
-                    choice
-                } else {
-                    return Err(ValidationError::UnsupportedPrivateVotes.into());
-                };
+                let address =
+                    Identifier::from(identifier).to_address(chain_addr::Discrimination::Production);
 
                 let address: Address = address.into();
                 let wallet = self
                     .wallets
                     .get_mut(&address)
                     .ok_or_else(|| Error::NonVotingAccount(address.to_string()))?;
 
-                let vote_plan = self
-                    .voteplans
-                    .get(vote_cast.vote_plan())
-                    .ok_or(Error::MissingVoteplanError)?;
-                let proposals_idx = vote_cast.proposal_index();
-
                 // we still use the old block0 hash because the new ledger will still use the old one for
                 // verifications. This makes possible the usage of old VoteTally transactions without the need
                 // to be replayed.
-                debug!(
-                    "replaying vote from {}, vote plan: {}, proposal idx: {}, choice: {:?}",
-                    identifier,
-                    &vote_plan.to_id(),
-                    proposals_idx,
-                    &choice
-                );
-                let res = wallet
-                    .issue_vote_cast_cert(
-                        &self.old_block0_hash,
-                        &self.fees,
-                        vote_plan,
-                        proposals_idx,
-                        choice,
-                    )
-                    .unwrap();
+                let res = FragmentFactory::new(self.old_block0_hash.into_hash(), self.fees)
```

Comment:
I'd say we should move this to the core API, or maybe improve the regular tx builder so that it's as convenient to use as the testing stuff.

Reply:
Part of the reasons it's more convenient to use this is that it relies on other testing stuff like `Wallet`, which we were using previously anyway. AFAIK we don't have the concept of a wallet in jormungandr production code, thus making it more difficult to streamline such operations and/or "promote" `FragmentFactory` to the core API.

Maybe we could see if chain-wallet-libs can offer a a valid option for a production ready wallet which is also easy enough to work with.